curation_helper_scripts/renaming_files_without_re-conversion/01_parse_dcm2bids_inventory.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+import re
+import subprocess
+import argparse
+
+import pandas as pd
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Creates text files with each series description as the filename. "
+                                                 "These files contain a list of scan names associated with "
+                                                 "the series description.")
+
+    parser.add_argument("--input_dir", type=Path, action='store', dest='inputdir', help="Path to input directory.")
+    parser.add_argument("--output_dir", type=Path, action='store', dest='outdir', help="Path to output directory.")
+    parser.add_argument("--series_desc_list", type=Path, action='store', dest='outdir', help="Path to output directory.")
+
+    return parser.parse_args()
+
+
+def run_shell_cmd(cmdstr):
+    pipe = subprocess.Popen(cmdstr, bufsize=1, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                            universal_newlines=True, encoding='utf8', shell=True)
+    return pipe.stdout
+
+
+def main():
+    root = Path('/Users/arshitha/Desktop/OneDesk/Projects/17M081_June_2022')  # directory with bids formatted data and code files
+    data = root.joinpath('17M081_jsons')  # bids dataset
+    series_descriptions = root.joinpath('all_series_descriptions.txt')  # file with list of all unique series descriptions
+
+    # hardcoded
+    desc_with_orig = ['ADNI2 2D FLAIR', 'Anat T1w MP-RAGE 1mm (ABCD)', 'Anat T2w CUBE (ABCD)', 'Axial T2 2D FLAIR ADNI2', 'Optional - Anat T2w CUBE FLAIR','Sag_MPRAGE_T1']
+    desc_with_optional = ['Sagittal 3D FLAIR']
+    desc_list = pd.read_csv(series_descriptions, sep='\n', header=None)[0].to_list()
+
+    cmds_list = []
+    for description in desc_list:
+        modified_description = re.sub(r"[\([{})\]]", "", description)
```

Comment:
Is the regular expression substitution to make it just alphanumerics?  If so, this is my favorite function for that:

```python
# convert words and strings to exclusively alphanumerics for BIDS' sake
def bidsify(input):
    return re.sub(r'[\W_]+', '', input)
```

Reply:
This didn't work as intended so I'm keeping what I have in the scripts.